src/main/java/com/faforever/client/vault/replay/ReplayVaultController.java

```diff
@@ -104,8 +114,22 @@ public void initialize() {
 
     durationColumn.setCellValueFactory(this::durationCellValueFactory);
     durationColumn.setCellFactory(this::durationCellFactory);
+  }
+
+  @Override
+  protected void onDisplay(NavigateEvent navigateEvent) {
+    if (isDisplayingForFirstTime) {
+      loadLocalReplaysInBackground();
+      isDisplayingForFirstTime = false;
+    }
 
-    loadLocalReplaysInBackground();
+    super.onDisplay(navigateEvent);
+  }
+
+  protected void loadLocalReplaysInBackground() {
+    // TODO: Display information that view contents are loading
```

Comment:
Is something "blocking" us to not to have TODO? I mean, if we leave that todo here, this will stay for ages :-)

Reply:
:D 
I do believe that small incremental improvements are better than big changesets. That said, I'd promise to look at that TODO once the PR is merged. :)

Reply:
Fixed.